releasenotes/notes/fix-pipeline-components-waiting-edge-cases-51a3054e3f1adb6e.yaml

```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Added a fix to the pipeline's component scheduling alogrithm to reduce edge cases where the execution order of
+    components that are simultaneously waiting for inputs has an impact on a pipeline's output.
```

Comment:
Maybe we could add how the tie break is made. First the topological sort then the lexicographical vs. just lexicographical from before.